src/azure-cli-core/azure/cli/core/commands/template_create.py

```diff
@@ -110,3 +122,121 @@ def handle_folding(namespace):
 
     return handle_folding
 
+# NEW STYLE
+
+# pylint: disable=line-too-long
+def get_folded_parameter_help_string(
+        display_name, allow_none=False, allow_new=False, other_required_option=None):
+    """ Assembles a parameterized help string for folded parameters. """
+    quotes = '""' if platform.system() == 'Windows' else "''"
+
+    if not allow_none and not allow_new:
+        help_text = 'Name or ID of an existing {}.'.format(display_name)
+    elif not allow_new and allow_none:
+        help_text = 'Name or ID of an existing {}, or {} for none.'.format(display_name, quotes)
+    elif allow_new and not allow_none:
+        help_text = 'Name or ID of the {}. Will create resource if it does not exist.'.format(display_name)
+    elif allow_new and allow_none:
+        help_text = 'Name or ID of the {}, or {} for none. Will create resource if it does not exist.'.format(display_name, quotes)
+
+    # add parent name option string (if applicable)
+    if other_required_option:
+        help_text = '{} If name specified, also specify {}'.format(help_text, other_required_option)
+    return help_text
+
+def _validate_name_or_id(
+        resource_group_name, property_value, property_type, parent_value, parent_type):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    has_parent = parent_type is not None
+    if is_valid_resource_id(property_value):
+        resource_id_parts = parse_resource_id(property_value)
+        value_supplied_was_id = True
+    elif has_parent:
```

Comment:
Correct. This logic is intended to help with the fat client parameters. Currently, none of those require grandparent logic. If we eventually did, this would likely be refactored to accept a collection of other required options. Currently all fat client resources are either independent or have, at most, a parent.